code/modules/reagents/chemistry/reagents/drug_reagents.dm

```diff
@@ -570,3 +570,94 @@
 		M.Jitter(4)
 		M.Dizzy(4)
 	..()
+
+
+///Can bring a corpse back to life temporarily (if heart is intact)
+///And prevents dying
+/datum/reagent/drug/nooartrium
+	name = "Nooartrium"
+	description = "A reagent that is known to stimulate the heart in a dead patient, temporarily bringing back recently dead patients at great cost to their heart."
+	overdose_threshold = 25
+	metabolization_rate = 0.2 * REM
+	self_consuming = TRUE //No pesky liver shenanigans
+	///If we brought someone back from the dead
+	var/back_from_the_dead = FALSE
+
+
+/datum/reagent/drug/nooartrium/on_mob_add(mob/living/affected_mob)
+	if(affected_mob.suiciding)
+		return
+	var/obj/item/organ/heart/heart = affected_mob.getorganslot(ORGAN_SLOT_HEART)
+	if(!heart || heart.organ_flags & ORGAN_FAILING)
+		return
+	ADD_TRAIT(affected_mob, TRAIT_NOCRITDAMAGE, FROM_NOOARTRIUM)
+	ADD_TRAIT(affected_mob, TRAIT_NODEATH, FROM_NOOARTRIUM)
+	ADD_TRAIT(affected_mob, TRAIT_NOHARDCRIT, FROM_NOOARTRIUM)
+	ADD_TRAIT(affected_mob, TRAIT_NOSOFTCRIT, FROM_NOOARTRIUM)
+	ADD_TRAIT(affected_mob, TRAIT_STABLEHEART, FROM_NOOARTRIUM)
+	ADD_TRAIT(affected_mob, TRAIT_NOLIMBDISABLE, FROM_NOOARTRIUM)
```

Comment:
```suggestion
```

Immunity to crits and stuns is one thing, but this doesn't suddenly reinforce their limbs with adamantium too. 

This should make powerful super zombies, but not actually unstoppable zombies. Leave the ability to disable them one limb at a time at least or there is no counterplay available at all.

Reply:
Having tested with it some now I would like to reiterate this point. 

Your testing is flawed, dead players do not automatically have disabled limbs, though there is a chance they may have some.